crates/solver/src/settlement_submission.rs

```diff
@@ -82,23 +80,25 @@ impl SolutionSubmitter {
             TransactionStrategy::ArcherNetwork {
                 archer_api,
                 max_confirm_time,
+                retry_interval,
             } => {
-                let submitter = ArcherSolutionSubmitter::new(
+                let submitter = Submitter::new(
                     &self.web3,
                     &self.contract,
                     &account,
                     archer_api,
                     self.gas_price_estimator.as_ref(),
-                    self.gas_price_cap,
                 )?;
-                let result = submitter
-                    .submit(
-                        self.target_confirm_time,
-                        SystemTime::now() + *max_confirm_time,
-                        settlement,
-                        gas_estimate,
-                    )
-                    .await;
+                let params = SubmitterParams {
+                    target_confirm_time: self.target_confirm_time,
+                    gas_estimate,
+                    gas_price_cap: self.gas_price_cap,
+                    deadline: Some(Instant::now() + *max_confirm_time),
+                    pay_gas_to_coinbase: Some(U256::from(18346)),
+                    additional_miner_tip: None,
```

Comment:
Instead of implementing this as a submission parameter, couldn't we instead:
```rust
struct FlashbotsGasEstimator {
    additional_tip: f64,
    inner: Arc<dyn GasEstimating>,
}
impl GasEstimating for FlashbotsGasEstimator {
    async fn estimate_with_limits(
        &self,
        gas_limit: f64,
        time_limit: Duration,
    ) -> Result<EstimatedGasPrice> {
        let mut gas_price = self.inner.estimate_with_limits(gas_limit, time_limit).await?;
        if let Some(eip1559) = &mut gas_price.eip1559 {
            eip1559.max_fee_per_gas += self.additional_tip;
            eip1559.max_priority_fee_per_gas += self.additional_tip;
        } else {
            gas_price.legacy += self.additional_tip;
        }
        Ok(gas_price)
    }
}
```

And then when creating the submitter for `TransactionStrategy::Flashbots` we just pass in:
```rust
FlashbotsGasEstimator {
    addition_tip: flashbots_tip,
    inner: self.gas_price_estimator.clone(),
}
```

Reply:
Nice one! Now `additional_tip` and `gas_price_cap` are coupled with the gas price estimator, as they should.